src/std/rpc/mod.rs

```diff
@@ -55,3 +57,59 @@ pub struct ReadProof<Hash> {
     /// A proof used to prove that storage entries are included in the storage trie
     pub proof: Vec<sp_core::Bytes>,
 }
+
+// Based on structures here
+// https://github.com/paritytech/substrate/blob/cf086eeb894e61b45c732b3dad2ce4dcc54fa1af/frame/transaction-payment/src/types.rs#L68
+// Would require pallet-transaction-payment and sp-rpc (for NumberOrHex) since the RPC returns either a number or hex string
+#[derive(Debug, Serialize, Deserialize)]
```

Comment:
Have you checked the amount of extra packages that are included? From my experience, the substrate crates have a lot of overlapping dependencies, so it might not matter that much if we include them.

Reply:
I just looked and there doesn't seem to be anything crazy...I will test it again on my end.

Reply:
Done, with feature-gate for staking